src/thor/route_matcher.cc

```diff
@@ -407,15 +407,15 @@ bool RouteMatcher::FormPath(const std::shared_ptr<DynamicCost>* mode_costing,
           // get the cost of traversing the node and the remaining part of the edge
           auto& costing = mode_costing[static_cast<int>(mode)];
           nodeinfo = end_edge_tile->node(n->first);
-          elapsed +=
-              costing->TransitionCost(end_de, nodeinfo, prev_edge_label) +
-              costing->EdgeCost(end_de, end_edge_tile, second_of_week) * end_edge.percent_along();
+          auto transition_cost = costing->TransitionCost(end_de, nodeinfo, prev_edge_label);
+          elapsed += transition_cost + costing->EdgeCost(end_de, end_edge_tile, second_of_week) *
+                                           end_edge.percent_along();
           // overwrite time with timestamps
           if (options.use_timestamps())
             elapsed.secs = shape.back().epoch_time() - shape[0].epoch_time();
 
           // Add end edge
-          path_infos.emplace_back(mode, elapsed.secs, end_edge_graphid, 0, elapsed.cost, -1);
+          path_infos.emplace_back(mode, elapsed, end_edge_graphid, 0, -1, transition_cost);
```

Comment:
this was a small bug. wasnt storing the transition cost separately for simple shape matching